# FinTrack — Personal Budget Tracker 
**Course:** Data Analytics  |  **Date:** 5/11/2026 
 
This notebook collects monthly income and expenses, then 
produces a formatted budget summary — no libraries required.

## Section 1: Analyst information 
Enter your name and the month being analysed when prompted.

In [1]:
analyst = input("Analyst name:  ")
month   = input("Budget month:  ")

Analyst name:   Dania
Budget month:   May 2026


## Section 2: Monthly income and expenses 
Enter each dollar amount when prompted. Values are cast to float 
so arithmetic works correctly — no libraries needed.

In [2]:
income        = float(input("Monthly income $:    "))
rent          = float(input("Rent / housing $:    "))
food          = float(input("Food / groceries $:  "))
transport     = float(input("Transport $:         "))
utilities     = float(input("Utilities $:         "))
entertainment = float(input("Entertainment $:     "))

print(f"\nrent_per          = {(rent / income) * 100:.2f}")
print(f"food_per          = {(food / income) * 100:.2f}")
print(f"transport_per     = {(transport / income) * 100:.2f}")
print(f"utilities_per     = {(utilities / income) * 100:.2f}")
print(f"entertainment_per = {(entertainment / income) * 100:.2f}")

Monthly income $:     4500
Rent / housing $:     1200
Food / groceries $:   380
Transport $:          150
Utilities $:          95
Entertainment $:      30



rent_per          = 26.67
food_per          = 8.44
transport_per     = 3.33
utilities_per     = 2.11
entertainment_per = 0.67


## Section 3: Budget calculations 
Total expenses, remaining budget, and savings rate are computed
using plain Python arithmetic — no import statements required.

In [3]:
total_expenses = rent + food + transport + utilities + entertainment
remaining      = income - total_expenses
savings_rate   = (remaining / income) * 100

In [4]:
if savings_rate >= 20.0:
    print("Healthy")
elif savings_rate >= 10.0:
    print("Tight")
else:
    print("Over budget")

Healthy


| expense       | % of income           |
| ------------  | --------------------- |
| rent          | 26.67 %               |
| food          | 8.44 %                |
| transport     | 3.33 %                |
| utilities     | 2.11 %                |
| entertainment | 0.67 %                |

## Section 4: Budget summary report
The report below is generated using Python f-strings.

In [5]:
print(f""" 
====== FinTrack Budget Report ====== 
Analyst  : {analyst} 
Month    : {month} 
------------------------------------ 
Income         : ${income:,.2f} 
Rent/Housing   : ${rent:,.2f} 
Food/Groceries : ${food:,.2f} 
Transport      : ${transport:,.2f} 
Utilities      : ${utilities:,.2f} 
Entertainment  : ${entertainment:,.2f}
------------------------------------ 
Total expenses : ${total_expenses:,.2f} 
Remaining      : ${remaining:,.2f} 
Savings rate   : {savings_rate:.1f}% 
==================================== 
""")

 
====== FinTrack Budget Report ====== 
Analyst  : Dania 
Month    : May 2026 
------------------------------------ 
Income         : $4,500.00 
Rent/Housing   : $1,200.00 
Food/Groceries : $380.00 
Transport      : $150.00 
Utilities      : $95.00 
Entertainment  : $30.00
------------------------------------ 
Total expenses : $1,855.00 
Remaining      : $2,645.00 
Savings rate   : 58.8% 

